### DISCUSSION TITLE
How do you add Y axis labels and names?

### DISCUSSION THREAD

**Discussion by jazzmonger** [dbuezas/lovelace-plotly-graph-card#249] [Q&A] [answered]
is there a way to name the y axis labels on the left AND rightside?

%is on the left automatically, but I'd like to label it with text.  And no way I could find to label the right Y axis...
![image](https://user-images.githubusercontent.com/52110065/218557981-1440b800-1f90-4c13-871e-5e6e01ffc45f.png)

**Comment by dbuezas** [comment, marked as answer]
Yes you have two options:
1. Give the entitiy that uses the right axis a `unit_of_measurement`. This usually comes from home_assistant, so I assume you are plotting an attribute. By default, this card will use that as the axis title. You should post your yaml too :)
2. Set the yaxis2 title manually:
```yaml
layout:
  yaxis2:
    title: "I'm the right axis!"
```
More examples [here](https://plotly.com/javascript/axes/) and the full API [here](https://plotly.com/javascript/reference/layout/yaxis/#layout-yaxis)

Pro tip: googling "how to do xyz in plotlyjs" will usually get you what you want when styling your plot, and may also inspire some new ideas too :)

**Comment by dbuezas** [comment]
Also, give the [custom range selector](https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/231#discussioncomment-4869001) a try, it's a more advanced technique, but it gives better results

**Comment by jazzmonger** [comment]
title: " " works great!

Thanks 

> On Feb 13, 2023, at 1:42 PM, David Buezas ***@***.***> wrote:
> 
> title: "

-- 


This message may contain confidential and/or privileged information. If 
you are not the addressee or are not authorized to receive this for the 
addressee, you must not use, copy, disclose, or take any action based on 
this message or any information herein. If you have received this message 
in error, please advise the sender immediately by reply email and delete 
this message.
